src/librustdoc/core.rs

```diff
@@ -129,6 +129,7 @@ pub fn run_core(search_paths: SearchPaths, cfgs: Vec<String>, externs: Externs,
     let sess = session::build_session_(sessopts, cpath, diagnostic_handler,
                                        codemap, cstore.clone());
     rustc_lint::register_builtins(&mut sess.lint_store.borrow_mut(), Some(&sess));
+    write::init_llvm(&sess);
```

Comment:
Hm ok this may be a *little* too much sprinkling around. Perhaps this could just be added to the constructor of a `Session`?